_plugins/jekyll_regex_match.rb

```
@@ -20,6 +20,35 @@ def regex_match_once(input_str, regex_str)
+          regex = Regexp.new(regex_pattern)  
+          items.select do |item|
+            next false unless item.is_a?(Hash) && item['date'].is_a?(String)
+
+            # Track cache size
+            @cache_size ||= 0
+            @cache_size += 1 unless item['cached_dates']
+
+            dates = item['cached_dates'] ||= item['date'].scan(regex).map(&:to_s) 
+            dates.include?(year)  
+          end
+          rescue RegexpError => e
```

rubocop

[warning] Useless assignment to variable - `e`.

(Lint/UselessAssignment)